### ISSUE TITLE
V4 doesn't compile - error with enqueBarrier()

### ISSUE THREAD

**Issue by tahmid95** [HPCE/hpce-2017-cw3#41]
When i try to compile v4 of the cw3 i get this error with enqueBarrier. I am using Ubuntu 16.04 with the Nvidia OpenCL SDK (I copied the CL folder from the SDK into the Include folder)

Has anyone come across this issue?

Running:
`g++ -I include -W -Wall -O3 -std=c++11 -o bin/mta16/step_world_v4_double_buffered src/mta16/step_world_v4_double_buffered.cpp src/heat.cpp  -lOpenCL
`

The error:
`src/mta16/step_world_v4_double_buffered.cpp: In function ‘void hpce::mta16::StepWorldV4DoubleBuffered(hpce::world_t&, float, unsigned int)’:
src/mta16/step_world_v4_double_buffered.cpp:158:10: error: ‘class cl::CommandQueue’ has no member named ‘enqueueBarrier’
    queue.enqueueBarrier(); // <- new barrier here
`

**Comment by JHertz5** [comment]
I had this same issue. The compiler is saying that enqueueBarrier is not defined in your OpenCL setup. After looking at 'opencl-sdk/include/CL/cl.hpp' (or wherever cl.hpp is for you), I saw that I was using a cl.hpp file for "CL Version 1.2" and enqueueBarrier was included in the file, but was in a section that looked like this
![selection_017](https://cloud.githubusercontent.com/assets/22753502/20039999/f9d32d46-a446-11e6-9b5c-7b05e9991d56.png)

You can see in the #if defined part that enqueueBarrier will not be included unless explicitly told to by a #define in the the code that is using the library. It seems that enqueueBarrier is considered old code, and has been replaced with a function called enqueueBarrierWithWaitList. However, if we use these #defines to include enqueueBarrier, then we can still use it. 

TL;DR: For me, adding 
`#define CL_USE_DEPRECATED_OPENCL_1_1_APIS`
near the top of my code fixed this problem. I'd recommend having a look at cl.hpp to see whether your problem is caused by the same thing.

**Comment by zhiyb** [comment]
`test_opengl.cpp` do use
`#define CL_USE_DEPRECATED_OPENCL_1_1_APIS`
before including `CL/cl.hpp`

**Comment by tahmid95** [comment]
thank you, i realised this as well after hunting through the headers!
